src/App.tsx

```diff
@@ -1,61 +1,127 @@
 import './App.scss';
+import React, { useState } from 'react';
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+import usersFromServer from './api/users';
+import todosFromServer from './api/todos';
+import { TodoWithUser, User } from './App.types';
+import { TodoList } from './components/TodoList';
+
+const todosWithUser: TodoWithUser[] = todosFromServer.map(
+  (todo): TodoWithUser => {
+    const user = usersFromServer.find(({ id }) => id === todo.userId) as User;
+
+    return {
+      ...todo,
+      user,
+    };
+  },
+);
+const defaultUserId = '0';
+const titleRegexp = /^[a-zA-Zа-яА-ЯёЁїЇіІєЄґҐ0-9\s]*$/;
+
+const validateTitle = (title: string): boolean => titleRegexp.test(title);
 
 export const App = () => {
+  const [todos, setTodos] = useState<TodoWithUser[]>([...todosWithUser]);
+  const [title, setTitle] = useState('');
+  const [userId, setUserId] = useState(defaultUserId);
+  const [hasTitleError, setHasTitleError] = useState(false);
+  const [hasUserError, setHasUserError] = useState(false);
+  const isNotSelectedUser = userId === defaultUserId;
```

Comment:
when we're dealing with booleans, we usually make the logic from positive, e.g. `isSelectedUser`, and use with `NOT (!)` operator.

It is difficult to read stuff like `!isNotSelectedUser`, since I have to compile in my mind what this variable means. In my mind it would sound like `not is not selected user`, this is difficult to read